src/main/java/database/LRUCache.java

```diff
@@ -0,0 +1,34 @@
+package database;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class LRUCache<K, V> extends LinkedHashMap<K, V> {
+
+    private final int bufferSize;
+    private final PageManager pageManager;
+
+    public LRUCache(int initialCapacity, float loadFactor, boolean accessOrder, PageManager pageManager) {
+        super(initialCapacity, loadFactor, accessOrder);
+        this.pageManager = pageManager;
+        this.bufferSize = initialCapacity;
+    }
+
+    @Override
+    protected boolean removeEldestEntry(Map.Entry<K, V> eldest) {
+        boolean isBufferPoolOverCapacity = size() > bufferSize;
+        boolean isUnpinned = !((Page) eldest.getValue()).isPinned();
+
+        if (isBufferPoolOverCapacity && isUnpinned) {
+            flushIfDirty((PageId) eldest.getKey(), (Page) eldest.getValue());
+            return true;
+        }
+        return false;
+    }
+
+    private void flushIfDirty(PageId pageId, Page page) {
```

Comment:
```suggestion
    private void flush(PageId pageId, Page page) {
```

해당 메서드를 사용하는 측에서는 더티 페이지의 여부가 중요하나요?